taskana-adapter-camunda-tasklistener/src/main/java/pro/taskana/adapter/camunda/tasklistener/TaskanaTaskListener.java

```diff
@@ -160,92 +170,63 @@ private void prepareAndExecuteStatement(Connection connection, DelegateTask dele
         }
     }
 
-    private String getReferencedTaskJson(DelegateTask delegateTask) {
-
-        StringBuilder referencedTaskJsonBuilder = new StringBuilder();
-
-        referencedTaskJsonBuilder.append("{\"id\":\"")
-            .append(delegateTask.getId())
-            .append("\",\"created\":\"")
-            .append(formatDate(delegateTask.getCreateTime()))
-            .append("\",\"priority\":\"")
-            .append(delegateTask.getPriority());
-        if (delegateTask.getName() != null) {
-            referencedTaskJsonBuilder.append("\",\"name\":\"")
-                .append(delegateTask.getName().replace("\n", "\\n"));
-        }
-        if (delegateTask.getAssignee() != null) {
-            referencedTaskJsonBuilder.append("\",\"assignee\":\"")
-                .append(delegateTask.getAssignee().replace("\n", "\\n"));
-        }
-        if (delegateTask.getDueDate() != null) {
-            referencedTaskJsonBuilder.append("\",\"due\":\"")
-                .append(formatDate(delegateTask.getDueDate()));
-        }
-        if (delegateTask.getDescription() != null) {
-            referencedTaskJsonBuilder.append("\",\"description\":\"")
-                .append(delegateTask.getDescription().replace("\n", "\\n"));
-        }
-        if (delegateTask.getOwner() != null) {
-            referencedTaskJsonBuilder.append("\",\"owner\":\"")
-                .append(delegateTask.getOwner().replace("\n", "\\n"));
-        }
-        if (delegateTask.getTaskDefinitionKey() != null) {
-            referencedTaskJsonBuilder.append("\",\"taskDefinitionKey\":\"")
-                .append(delegateTask.getTaskDefinitionKey());
-        }
-        String classificationKey = getUserTaskExtensionProperty(delegateTask, "classification-key");
-        if (classificationKey != null) {
-            referencedTaskJsonBuilder.append("\",\"classificationKey\":\"")
-                .append(classificationKey);
-        }
-        String domain = getProcessModelExtensionProperty(delegateTask, "domain");
-        if (domain != null) {
-            referencedTaskJsonBuilder.append("\",\"domain\":\"")
-                .append(domain);
-        }
-
-        referencedTaskJsonBuilder.append("\",\"variables\":\"{")
-            .append(getProcessVariables(delegateTask));
+    private String getReferencedTaskJson(DelegateTask delegateTask) throws JsonProcessingException {
 
-        referencedTaskJsonBuilder.append("}");
-
-        return referencedTaskJsonBuilder.toString();
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
Please don't make a new ObjectMapper for each call, but use a member (or static) variable for the ObjectMapper. Please note, that initialization must be complete before ObjectMapper is used for the first time.

Reply:
Done